modules/lead/toolchain-ingress/main.tf

```diff
@@ -165,3 +165,58 @@ module "toolchain_ingress" {
   cluster_wide                    = true
   default_certificate             = "${var.namespace}/${module.toolchain_wildcard.cert_secret_name}"
 }
+
+// Ingress controller for internal access
+
+resource "kubernetes_service_account" "internal_nginx_ingress_service_account" {
+  metadata {
+    name      = "internal-nginx-ingress"
+    namespace = var.namespace
+  }
+  automount_service_account_token = true
+}
+
+resource "kubernetes_cluster_role_binding" "internal_nginx_ingress_role_binding" {
+  metadata {
+    name = "internal-nginx-ingress-binding"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.nginx_ingress_cluster_role.metadata[0].name
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = kubernetes_service_account.internal_nginx_ingress_service_account.metadata[0].name
+    namespace = kubernetes_service_account.internal_nginx_ingress_service_account.metadata[0].namespace
+  }
+}
+
+module "internal_wildcard" {
+  source = "../../common/certificates"
+
+  name      = "internal-wildcard"
+  namespace = var.namespace
+  domain    = "internal.${var.cluster_domain}"
+  enabled   = true
+
+  issuer_name = var.issuer_name
+  issuer_kind = var.issuer_kind
+}
+
+// Ingress for strictly allowing only Internal VPN traffic
+module "internal_ingress" {
+  source                          = "../../common/nginx-ingress"
+  namespace                       = var.namespace
+  name                            = "internal"
+  ingress_controller_type         = var.ingress_controller_type
+  ingress_external_traffic_policy = var.ingress_external_traffic_policy
+  ingress_class                   = "internal-nginx"
+  service_annotaitons = {
+    "service.beta.kubernetes.io/aws-load-balancer-internal": true
+  }
+  service_load_balancer_source_ranges = ["10.1.32.0/20"]
```

Comment:
the client cidr block is actually `10.1.48.0/20`: https://github.com/liatrio/aws-terraform/blob/master/environments/sharedsvc/network/internal-vpn.tf#L63

Reply:
I tested both... `10.1.32.0/20` works and `10.1.48.0/20` doesn't. I think we ran into this before and I think `10.1.48.0/20` is the address in shared services but the traffic is coming from a different ip range because of the cross account vpc configuration stuff. It is also documented as `VPC for internal VPN (split tunnel vpn peered to other subnets)` in the aws-terraform readme so i'm pretty sure it's the right one.